.github/workflows/build.yml

```diff
@@ -124,8 +124,8 @@ jobs:
         with:
           targets: "build,test,package"
           all: true
-          # Exclude E2E testing
-          args: "--exclude=hangar --configuration=release --output-style=stream --verbose"
+          # If pushing, exclude hangar because we're going to run it in a separate job
+          args: "${{ github.event_name = 'push' ?? '--exclude=hangar ' || '' }}--configuration=release --output-style=stream --verbose"
```

Comment:
Maybe we can tweak this slightly to model this in the NX config directly, i.e. `--configuration=prbuild` and `--configuration=release`.

Also can we make `github.event_name == 'push'` into a workflow scoped variable (`RUN_FULL_E2E_TESTS=...`)? I think it could be handy if we could tweak the conditions so we can e.g. opt into the full multi-platform test suite if we want by adding a GitHub label (doesn't have to be in this PR)

Reply:
> Also can we make github.event_name == 'push' into a workflow scoped variable

Should be able to do that, it's better semantically anyways 👍 

> Maybe we can tweak this slightly to model this in the NX config directly

Not sure if nx is meant to work that way (e.g. using configurations to conditionally run certain targets).

> opt into the full multi-platform test suite if we want by adding a GitHub label

For this and the previous point, I think this should be solved by actually having separate workflows.